shell/platform/android/io/flutter/view/AccessibilityBridge.java

```diff
@@ -35,35 +63,94 @@
+    @NonNull
     private final View decorView;
-    private Map<Integer, SemanticsObject> objects;
-    private Map<Integer, CustomAccessibilityAction> customAccessibilityActions;
+
+    // TODO(mattcarroll): what exactly is a semantic object, and why do we cache them?
+    @NonNull
+    private final Map<Integer, SemanticsObject> semanticsObjects;
+
+    // TODO(mattcarroll): what makes a CustomAccessibilityAction different than a regular
+    //                    AccessibilityAction. why do we cache them?
+    @NonNull
+    private final Map<Integer, CustomAccessibilityAction> customAccessibilityActions;
+
+    // TODO(mattcarroll): what is the source of truth for this supposed to be? Is it
+    //                    whether or not the OS has accessibility enabled? Or do other
+    //                    factors influence this value?
     private boolean accessibilityEnabled = false;
-    private SemanticsObject a11yFocusedObject;
+
+    // The widget within Flutter that currently has the focus of Android's
+    // accessibility system.
+    @Nullable
+    private SemanticsObject accessibilityFocusedObject;
+
+    // The widget within Flutter that currently has the focus of Android's input
+    // system.
+    // TODO(mattcarroll): is there a relationship between this and the InputMethodManager?
+    @Nullable
     private SemanticsObject inputFocusedObject;
+
+    // The widget within Flutter that currently sits beneath a cursor, e.g,
+    // beneath a stylus or mouse cursor.
+    @Nullable
     private SemanticsObject hoveredObject;
+
     private int previousRouteId = ROOT_NODE_ID;
-    private List<Integer> previousRoutes;
+
+    @NonNull
+    private final List<Integer> previousRoutes;
+
+    @NonNull
     private Integer lastLeftFrameInset = 0;
 
+    // Handler for all messages received from Flutter via the {@code accessibilityChannel}
     private final AccessibilityChannel.AccessibilityMessageHandler accessibilityMessageHandler = new AccessibilityChannel.AccessibilityMessageHandler() {
+        /**
+         * The Dart application would like the given {@code message} to be announced.
+         */
         @Override
         public void announce(@NonNull String message) {
-            owner.announceForAccessibility(message);
+            rootAccessibilityView.announceForAccessibility(message);
         }
 
+        /**
+         * The user has tapped on the widget with the given {@code nodeId}.
+         */
         @Override
         public void onTap(int nodeId) {
             sendAccessibilityEvent(nodeId, AccessibilityEvent.TYPE_VIEW_CLICKED);
         }
 
+        /**
+         * The user has long pressed on the widget with the given {@code nodeId}.
+         */
         @Override
         public void onLongPress(int nodeId) {
             sendAccessibilityEvent(nodeId, AccessibilityEvent.TYPE_VIEW_LONG_CLICKED);
         }
 
+        /**
+         * The user has opened a popup window, menu, dialog, etc.
```

Comment:
Just tooltip

Reply:
Are you sure?  I got that either from the JavaDoc or the Android source code, I can't remember which one.  But I definitely looked it up.

Reply:
Positive: https://github.com/flutter/flutter/search?utf8=%E2%9C%93&q=TooltipSemanticsEvent&type=